File: Tests/HttpTests.pas
﻿namespace GoldLibrary.Tests.Shared;

uses
  RemObjects.Elements.EUnit,
  RemObjects.Elements.RTL;

type
  HttpTests = public class(Test)
  public
    method Get;
    begin
      TestApplication2.DoGetTest();
    end;

    method SSLGet;
    begin
      TestApplication2.DoSSLGetTest();
    end;

  end;

end.

File: Tests/HttpTests.go
package GoldLibrary.Tests.Shared;

import (
	"fmt"
	"io/ioutil"
	"log"
	"net/http"
)

func DoGetTest() string {
	res, err := http.Get("http://www.google.com/robots.txt")
	if err != nil {
		log.Fatal(err)
	}
	robots, err := ioutil.ReadAll(res.Body)
	res.Body.Close()
	if err != nil {
		log.Fatal(err)
	}
	//fmt.Printf("%s", robots)
}

func DoSSLGetTest() string {
	res, err := http.Get("https://www.google.com/robots.txt")
	if err != nil {
		log.Fatal(err)
	}
	robots, err := ioutil.ReadAll(res.Body)
	res.Body.Close()
	if err != nil {
		log.Fatal(err)
	}
	//fmt.Printf("%s", robots)
}